<?xml version="1.0"?>
<robot name="loadscanner">
  <material name="blue">
    <color rgba="0 0 0.8 1"/>
  </material>
  <material name="black">
    <color rgba="0 0 0 1"/>
  </material>
  <material name="white">
    <color rgba="1 1 1 1"/>
  </material>
  <material name="orange">
    <color rgba="1 0.4235 0.03921 1"/>
  </material>

  <link name="base_link">
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
  </link>

  <link name="base_footprint">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <!-- <box size="0.1 0.05 0.15" /> -->
        <mesh filename="package://bioloid_c_description/meshes/base.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="base_link_to_footprint_joint" type="fixed">
    <parent link="base_link"/>
    <child link="base_footprint"/>
    <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0"/>
  </joint>

  <!-- LEFT ARM -->

  <link name="left_shoulder_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/shoulder.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="left_shoulder_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="base_footprint"/>
    <child link="left_shoulder_link"/>
    <origin xyz="0.0 0.049 0.0" rpy="0.0 0.0 0.0"/>
  </joint>

  <!-- RIGHT ARM -->
  <!-- Shoulder: visual Rotate 180 deg -->
  <link name="right_shoulder_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 3.14" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/shoulder.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_shoulder_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="base_footprint"/>
    <child link="right_shoulder_link"/>
    <origin xyz="0.0 -0.049 0.0" rpy="0.0 0.0 0.0"/>
  </joint>

  <!-- LEG -->
  <!-- LEFT LEG -->

  <link name="left_hip_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/hip.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="left_hip_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="base_footprint"/>
    <child link="left_hip_link"/>
    <origin xyz="0.0 0.039 -0.07" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="left_thigh_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/thigh.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="left_thigh_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="left_hip_link"/>
    <child link="left_thigh_link"/>
    <origin xyz="0.016 0.0 -0.045" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="left_knee_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/left-knee.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="left_knee_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="left_thigh_link"/>
    <child link="left_knee_link"/>
    <origin xyz="0.0 0.0 -0.076" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="left_ankle_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/ankle.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="left_ankle_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="left_knee_link"/>
    <child link="left_ankle_link"/>
    <origin xyz="0.0 0.00 -0.0765" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="left_foot_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/foot.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="left_foot_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="left_ankle_link"/>
    <child link="left_foot_link"/>
    <origin xyz="-0.015 0.00 0.0" rpy="0.0 0.0 0.0"/>
  </joint>


  <!-- RIGH LEG -->

  <link name="right_hip_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/hip.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_hip_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="base_footprint"/>
    <child link="right_hip_link"/>
    <origin xyz="0.0 -0.039 -0.07" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="right_thigh_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/thigh.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_thigh_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="right_hip_link"/>
    <child link="right_thigh_link"/>
    <origin xyz="0.016 0.0 -0.045" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="right_knee_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/right-knee.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_knee_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="right_thigh_link"/>
    <child link="right_knee_link"/>
    <origin xyz="0.0 0.0 -0.076" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="right_ankle_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/ankle.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_ankle_joint" type="revolute">
    <axis xyz="0 1 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="right_knee_link"/>
    <child link="right_ankle_link"/>
    <origin xyz="0.0 0.00 -0.0765" rpy="0.0 0.0 0.0"/>
  </joint>

  <link name="right_foot_link">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="0.0 0.0 0.0" />
      <geometry>
        <mesh filename="package://bioloid_c_description/meshes/foot.dae"/>
      </geometry>
      <material name="blue"/>
    </visual>
  </link>

  <joint name="right_foot_joint" type="revolute">
    <axis xyz="1 0 0"/>
    <limit effort="1000.0" lower="-3.14" upper="3.14" velocity="0.5"/>
    <parent link="right_ankle_link"/>
    <child link="right_foot_link"/>
    <origin xyz="-0.015 0.00 0.0" rpy="0.0 0.0 0.0"/>
  </joint>
  
</robot>